TreeDB/docs/spec/storage-format.md

```
@@ -531,12 +531,19 @@ offsets []uint64  // row_count + 1, little-endian
+column-wide values section per offsets-list column. For multi-block parts, block
+payloads may use block-local offsets internally, but the image writer publishes a
+single global `row_count + 1` offsets array by dropping duplicate block starts and
+adding cumulative value bases; readers reconstruct block-local fallback payloads
+from those global sections. The offsets-list primitive validates exact offsets
+count, `offsets[0] == 0`, monotonic offsets, final offset equal to the value
+count, exact offsets/value byte lengths, Go `int` range before slicing,
+little-endian identity, and separate section metadata/checksums for offsets
+(8-byte elements) and values (4-byte elements). #1915 adds the safe writer and
+fallback reader into owned Go slices; unsafe direct-view readers,
+adapter-to-column_graph wiring, and graph search consumption remain deferred to
+#1916+.
```

markdownlint-cli2 (0.22.1)

[warning] No space after hash on atx style heading

(MD018, no-missing-space-atx)